lib/HunkView.ml

```diff
@@ -67,35 +91,91 @@ let ui_hunk_summary (hunk : string Patch.hunk) : Nottui.ui =
       at_symbols;
     ]
 
-let ui_unified_diff (hunk : string Patch.hunk) : Nottui.ui =
-  let hunk_summary = ui_hunk_summary hunk in
-  let hunk_content =
-    let blocks = Block.of_hunk hunk.Patch.lines in
-    let single_line_changes =
-      List.for_all
-        (function
-          | Block.Changed { mine; their; _ } ->
-              List.length mine = 1 && List.length their = 1
-          | _ -> true)
-        blocks
-    in
-    if single_line_changes then
-      let word_diff_blocks = List.map WordDiff.compute blocks in
-      let word_diff_lines = Block.to_hunk word_diff_blocks in
-      WordDiff.render_hunk_lines word_diff_lines
-    else WordDiff.render_hunk hunk
+let render_line_number (mine_num : int) (their_num : int)
+    (diff_type : [ `Added | `Removed | `Unchanged ]) (mode : rendering_mode) :
+    Ui.t =
+  match mode with
+  | TextMarkers -> (
+      match diff_type with
+      | `Added -> W.string (Printf.sprintf "   %2d + " (their_num + 1))
+      | `Removed -> W.string (Printf.sprintf "%2d    - " (mine_num + 1))
+      | `Unchanged ->
+          W.string (Printf.sprintf "%2d %2d   " (mine_num + 1) (their_num + 1)))
+  | Color -> (
```

Comment:
The only difference between these two match cases is that in one case `attr` is not passed ? These two cases should be factored to avoid a maintenance burden.

Here's a trick: It's possible to pass an `option` value as an optional argument:
```
let attr_added = None

W.string ?attr:attr_added "..."
```